src/Opaleye/Internal/Column.hs

```diff
@@ -10,12 +15,14 @@ import qualified Opaleye.Internal.HaskellDB.PrimQuery as HPQ
 --
 -- Do not use the 'Show' instance of 'Column'.  It will be deprecated
 -- in version 0.6.
-newtype Column pgType = Column HPQ.PrimExpr deriving Show
+data Column_ colType pgType = Column HPQ.PrimExpr deriving Show
+type Column = Column_ NonNullable
 
 -- | Only used within a 'Column', to indicate that it can be @NULL@.
 -- For example, a 'Column' ('Nullable' @PGText@) can be @NULL@ but a
 -- 'Column' @PGText@ cannot.
 data Nullable a = Nullable
+data NonNullable
```

Comment:
I don't understand how this would work since `Nullable` and `NonNullable` are of different kinds.  Shouldn't we have `data NonNullable a = NonNullable`?